src/Bicep.Core.IntegrationTests/Scenarios/ParamKeyVaultSecretReferenceTests.cs

```diff
@@ -0,0 +1,218 @@
+    {
+
+        [TestMethod]
+        public void ValidKeyVaultSecretReference()
+        {
+            var (template, diags, _) = CompilationHelper.Compile(
+                ("main.bicep", @"
+resource kv 'Microsoft.KeyVault/vaults@2019-09-01' existing = {
+  name: 'testkeyvault'
+}
+
+
+module secret 'secret.bicep' = {
+  name: 'secret'
+  params: {
+    mySecret: kv.getSecret('mySecret')
+  }
+}
+"),
+                ("secret.bicep", @"
+@secure()
+param mySecret string
+
+output exposed string = mySecret
+"));
+
+            diags.Should().BeEmpty();
+            template!.Should().NotBeNull();
+            var parameterToken = template!.SelectToken("$.resources[?(@.name == 'secret')].properties.parameters.mySecret")!;
+            using (new AssertionScope())
+            {
+                parameterToken.SelectToken("$.value")!.Should().BeNull();
+                parameterToken.SelectToken("$.reference.keyVault.id")!.Should().DeepEqual("[resourceId('Microsoft.KeyVault/vaults', 'testkeyvault')]");
+                parameterToken.SelectToken("$.reference.secretName")!.Should().DeepEqual("mySecret");
+                parameterToken.SelectToken("$.reference.secretVersion")!.Should().BeNull();
+            }
+        }
+
+        [TestMethod]
+        public void ValidKeyVaultSecretReferenceWithSecretVersion()
+        {
+            var (template, diags, _) = CompilationHelper.Compile(
+                ("main.bicep", @"
+resource kv 'Microsoft.KeyVault/vaults@2019-09-01' existing = {
+  name: 'testkeyvault'
+}
+
+module secret 'secret.bicep' = {
+  name: 'secret'
+  params: {
+    mySecret: kv.getSecret('mySecret','secretversionguid')
+  }
+}
+"),
+                ("secret.bicep", @"
+@secure()
+param mySecret string = 'defaultSecret'
+
+output exposed string = mySecret
+"));
+
+            diags.Should().BeEmpty();
+            template!.Should().NotBeNull();
+            var parameterToken = template!.SelectToken("$.resources[?(@.name == 'secret')].properties.parameters.mySecret")!;
+            using (new AssertionScope())
+            {
+                parameterToken.SelectToken("$.value")!.Should().BeNull();
+                parameterToken.SelectToken("$.reference.keyVault.id")!.Should().DeepEqual("[resourceId('Microsoft.KeyVault/vaults', 'testkeyvault')]");
+                parameterToken.SelectToken("$.reference.secretName")!.Should().DeepEqual("mySecret");
+                parameterToken.SelectToken("$.reference.secretVersion")!.Should().DeepEqual("secretversionguid");
+            }
+        }
+
+        [TestMethod]
+        public void InvalidKeyVaultSecretReferenceUsageInOutput()
+        {
+            var (template, diags, _) = CompilationHelper.Compile(
+                ("main.bicep", @"
+resource kv 'Microsoft.KeyVault/vaults@2019-09-01' existing = {
+  name: 'testkeyvault'
+}
+
+output exposed string = kv.getSecret('mySecret','secretversionguid')
+"));
+
+            diags.Should().NotBeEmpty();
```

Comment:
Mind adding a specific assertion for the diagnostics you're expecting (here and in other places)? It helps avoid accidental regressions in the error handling. See here for an example: https://github.com/Azure/bicep/blob/d3ea3e822d128d29b28101b54086c67bbfb7a10b/src/Bicep.Core.IntegrationTests/ScenarioTests.cs#L1237-L1244